src/modelfree/scheduling.py

```diff
@@ -1,31 +1,55 @@
 from abc import ABC, abstractmethod
+import collections
 import functools
+import itertools
+import operator
 
 
 class Scheduler(object):
     """Keep track of frac_remaining and return time-dependent values"""
-    def __init__(self, func_dict=None):
-        if func_dict is None:
-            func_dict = {}
-        assert isinstance(func_dict, dict)
-        self.func_dict = func_dict
-        self.allowed_func_types = ['lr', 'rew_shape', 'noise']
+    def __init__(self, annealer_dict=None):
+        if annealer_dict is None:
+            annealer_dict = {}
+        self.annealer_dict = annealer_dict
+        self.func_dict = {}
+        for k, v in annealer_dict.items():
+            self.func_dict[k] = v.get_value
+
+        self.conditionals = collections.defaultdict(lambda: False)
         self.frac_remaining = 1  # frac_remaining goes from 1 to 0
 
+    @staticmethod
```

Comment:
Google style guide recommends use of module-level private methods over `@staticmethod` and I tend to agree: http://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators

Reply:
I like that this was factored out into a function though.